include/gui_templates.h

```diff
@@ -31,37 +31,26 @@
 #include <QFont>
 #include <QDesktopWidget>
 
+// TODO: cleanup for qt6
+#if (QT_VERSION < QT_VERSION_CHECK(6,0,0))
+	#include <QScreen>
+#else
+	#include <QGuiApplication>
+#endif
+
 namespace lmms
 {
 
 
 // return DPI-independent font-size - font with returned font-size has always
 // the same size in pixels
-template<int SIZE>
-inline QFont pointSize( QFont _f )
+inline QFont pointSize(QFont _f, float SIZE)
 {
-	static const float DPI = 96;
-#ifdef LMMS_BUILD_WIN32
-	_f.setPointSizeF( ((float) SIZE+0.5f) * DPI /
-			QApplication::desktop()->logicalDpiY() );
-#else
-	_f.setPointSizeF( (float) SIZE * DPI /
-			QApplication::desktop()->logicalDpiY() );
-#endif
-	return( _f );
-}
-
+	// to calculate DPI of a screen to make it HiDPI ready
+	qreal devicePixelRatio = QGuiApplication::primaryScreen()->devicePixelRatio();
```

Comment:
When using `QGuiApplication::primaryScreen()` i think there could be some issues when Switching the App to an other Monitor with an diffrent DPI, We should grab the current Monitor for our Window

Reply:
I get what you mean but can you suggest a way to handle this one properly?

Reply:
I think this is something that should be handled by the GUI framework, i.e. by Qt. It just one example of why using these functions in application code is problematic.

Using code like this also won't do the "right thing" when LMMS is in the middle of being dragged from one screen to another with different DPI values. Qt likely knows that one part of the application rectangle must be rendered with different settings than the other and can do so. The application code does not even have the knowledge of the current state ("I am shown on two different monitors.") and cannot do the right thing.